src/ast/remove_globals.rs

```diff
@@ -25,88 +32,176 @@ use crate::{
 /// (set (x) 3)
 /// (Add (x) (x))
 /// ```
+///
+/// If later, this global is referenced in a rule:
+/// ```ignore
+/// (rule ((Neg y))
+///       ((Add x x)))
+/// ```
+/// We instrument the query to make the value available:
+/// ```ignore
+/// (rule ((Neg y)
+///        (= fresh_var_for_x (x)))
+///       ((Add fresh_var_for_x fresh_var_for_x)))
+/// ```
 pub(crate) fn remove_globals(
     type_info: &TypeInfo,
     prog: Vec<ResolvedNCommand>,
 ) -> Vec<ResolvedNCommand> {
+    let mut remover = GlobalRemover {
+        fresh: SymbolGen::new(),
```

Comment:
This seems dangerous, since the counters are not shared across `SymbolGen`s so you may accidentally get conflict names.

Reply:
Will fix soon! There are other problems with the symbolgen to address at the same time